src/vm/fieldmarshaler.cpp

```diff
@@ -1236,11 +1241,34 @@ BOOL IsStructMarshalable(TypeHandle th)
     return TRUE;
 }
 
-BOOL IsFixedBuffer(mdFieldDef field, IMDInternalImport *pInternalImport)
+BOOL IsFixedBuffer(MethodTable* pParentMT, mdFieldDef field, IMDInternalImport *pInternalImport, int* pNumElements, MethodTable** ppFixedBufferTypeMethodTable)
 {
-    HRESULT hr = pInternalImport->GetCustomAttributeByName(field, g_FixedBufferAttribute, NULL, NULL);
+    const void* pData;
+    ULONG cData;
 
-    return hr == S_OK ? TRUE : FALSE;
+    HRESULT hr = pInternalImport->GetCustomAttributeByName(field, g_FixedBufferAttribute, &pData, &cData);
+
+    if (hr == S_OK && cData != 0)
+    {
+        CustomAttributeParser ca(pData, cData);
+
+        CaArg args[2];
+        args[0].Init(SERIALIZATION_TYPE_TYPE);
+        args[1].Init(SERIALIZATION_TYPE_I4);
+
+        
+        if (FAILED(ParseKnownCaArgs(ca, args, lengthof(args)))) {
```

Comment:
Please use `ARRAYSIZE()`. I am trying to convert all array counting macro consistent.